Source/WebCore/platform/graphics/avfoundation/objc/SourceBufferPrivateAVFObjC.mm

```diff
@@ -136,9 +136,10 @@ - (void)dealloc
 - (void)invalidate
 {
     ASSERT(isMainThread());
-    auto protectedParent = _parent.get();
-    if (!protectedParent && !_layers.size() && !_renderers.size())
-        return;
+
+    [NSNotificationCenter.defaultCenter removeObserver:self];
+
+    _parent = nullptr;
```

Comment:
Is there a reason these two lines were moved up?

Reply:
in case removing any over the observer caused re-entrant code to try to read _parent